Sources/AsyncHTTPClient/AsyncAwait/HTTPClientRequest.swift

```diff
@@ -341,7 +341,13 @@ extension HTTPClientRequest.Body {
         public static let unknown: Self = .init(storage: .unknown)
 
         /// The size of the request body is known and exactly `count` bytes
+        @_disfavoredOverload
         public static func known(_ count: Int) -> Self {
```

Comment:
I think we have to deprecate.

The number of times this is called with an integer literal is actually very low. This will usually be called with a computed value, which will be of type `Int`. That means the issue isn't actually at this location: the user failed to even _compute_ the right size.

We need to push users over to computing on sizes in `Int64`.

One option, if we wanted to do it, would be to confine the deprecation to the 32-bit platforms (which we can do by a series of clever games with `#if`). But I have the voice of @weissi in my head telling me that that's a bad idea because it produces warnings that most libraries will never see, meaning they won't actually address their issues. So I guess I tentatively favour deprecation.

Reply:
+1 on deprecation. I think it's beneficial:

- As @Lukasa points out: the use cases for `.known(...)` are limited -> blast radius of people who we annoy needlessly is small
- We do want the people who use `.known(Int)` to migrate